### PR TITLE
Drop non-dimensional coordinates in Compositor

### LINKED ISSUE #795: Composites delayed in the presence of non-dimensional coordinates
**Describe the bug**
Creating composites is delayed if the datasets have non-dimensional coordinates (`acq_time` in this case). The reason is that `xr.concat` in `GenericCompositor._concat_datasets` fails because

a) Channel differences don't have an `acq_time` coordinate
b) The `acq_time` coordinates are not equal across the channels

**To Reproduce**
a)
```python
from satpy import Scene
from satpy.utils import logging_on
import logging
import glob
logging_on(logging.DEBUG)
filenames = glob.glob('tline/data/*201903011200*')
scene = Scene(filenames=filenames, reader='seviri_l1b_hrit')
scene.load(['airmass'])
scene.show('airmass')
```

b)
```python
scene.load(['overview'])
scene.show('overview')
```

**Expected behavior**
The composites to be displayed

**Actual results**
Tracebacks & logging (first line is the error message from `xr.concat`)

a)
<details>

```
[DEBUG: 2019-05-28 15:18:42 : satpy.composites] Original exception for incompatible areas: encountered unexpected variable 'acq_time'
[DEBUG: 2019-05-28 15:18:42 : satpy.scene] Delaying generation of DatasetID(name='airmass', wavelength=None, resolution=None, polarization=None, calibration=None, level=None, modifiers=None) because of incompatible areas
[WARNING: 2019-05-28 15:18:42 : satpy.scene] The following datasets were not created and may require resampling to be generated: DatasetID(name='airmass', wavelength=None, resolution=None, polarization=None, calibration=None, level=None, modifiers=None)
[DEBUG: 2019-05-28 15:18:42 : satpy.scene] Unloading dataset: DatasetID(name='IR_108', wavelength=(9.8, 10.8, 11.8), resolution=3000.403165817, polarization=None, calibration='brightness_temperature', level=None, modifiers=())
[DEBUG: 2019-05-28 15:18:42 : satpy.scene] Unloading dataset: DatasetID(name='WV_073', wavelength=(6.85, 7.35, 7.85), resolution=3000.403165817, polarization=None, calibration='brightness_temperature', level=None, modifiers=())
[DEBUG: 2019-05-28 15:18:42 : satpy.scene] Unloading dataset: DatasetID(name='IR_097', wavelength=(9.38, 9.66, 9.94), resolution=3000.403165817, polarization=None, calibration='brightness_temperature', level=None, modifiers=())
Traceback (most recent call last):
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/readers/__init__.py", line 335, in __getitem__
    return super(DatasetDict, self).__getitem__(item)
KeyError: 'airmass'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 11, in <module>
    scene.show('airmass')
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/scene.py", line 1148, in show
    img = get_enhanced_image(self[dataset_id].squeeze(), overlay=overlay)
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/scene.py", line 680, in __getitem__
    return self.datasets[key]
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/readers/__init__.py", line 337, in __getitem__
    key = self.get_key(item)
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/readers/__init__.py", line 326, in get_key
    best=best, **dfilter)
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/readers/__init__.py", line 277, in get_key
    raise KeyError("No dataset matching '{}' found".format(str(key)))
KeyError: "No dataset matching 'DatasetID(name='airmass', wavelength=None, resolution=None, polarization=None, calibration=None, level=None, modifiers=None)' found"
```

</details>

b)
<details>

```
[DEBUG: 2019-05-28 15:21:49 : satpy.composites] Original exception for incompatible areas: variable acq_time not equal across datasets
[DEBUG: 2019-05-28 15:21:49 : satpy.scene] Delaying generation of DatasetID(name='overview', wavelength=None, resolution=None, polarization=None, calibration=None, level=None, modifiers=None) because of incompatible areas
[WARNING: 2019-05-28 15:21:49 : satpy.scene] The following datasets were not created and may require resampling to be generated: DatasetID(name='overview', wavelength=None, resolution=None, polarization=None, calibration=None, level=None, modifiers=None)
Traceback (most recent call last):
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/readers/__init__.py", line 335, in __getitem__
    return super(DatasetDict, self).__getitem__(item)
KeyError: 'overview'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 11, in <module>
    scene.show('overview')
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/scene.py", line 1148, in show
    img = get_enhanced_image(self[dataset_id].squeeze(), overlay=overlay)
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/scene.py", line 680, in __getitem__
    return self.datasets[key]
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/readers/__init__.py", line 337, in __getitem__
    key = self.get_key(item)
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/readers/__init__.py", line 326, in get_key
    best=best, **dfilter)
  File "/cmsaf/nfshome/sfinkens/software/devel/satpy/satpy/readers/__init__.py", line 277, in get_key
    raise KeyError("No dataset matching '{}' found".format(str(key)))
KeyError: "No dataset matching 'DatasetID(name='overview', wavelength=None, resolution=None, polarization=None, calibration=None, level=None, modifiers=None)' found"
```

</details>

**Screenshots**

**Environment Info:**
 - OS: Linux
 - Satpy Version: '0.14.2+198.gcd65262b'
 - PyResample Version:

### PR DESCRIPTION
**Pull request by sfinkens**
Attempt to fix #795 by dropping non-dimensional coordinates.

 - [X] Closes #795 
 - [x] Tests added and test suite added to parent suite 
 - [x] Tests passed 
 - [x] Passes ``flake8 satpy``

### CODE REVIEW

**Comment by BENR0** [comment]
I've come across this too a couple of days ago. Just calling resample after loading fixes this. I don't know how resample handles 2d coords but a similar approach of handling coords could maybe be used here.
Alternatively the proposed fix maybe could include a check for 2d coords (and any other non dim coords which should be kept and is the same over all datasets) demote those to variables, drop every other coord and promote them to coords again afterwards.

**Comment by sfinkens** [comment]
I've implemented @djhoese's suggestion to exclude "neglible" non-dimensional coordinates (defined in a global list)

**Comment by djhoese** [comment]
Wasn't the original problem for this PR the `acq_time` coordinate? You aren't excluding that here, are you? Also, what does PEP257 say about `"""` comments?

**Comment by sfinkens** [comment]
@djhoese As you suggested all non-dimensional coordinates with `time` in their name are excluded. PEP257 says, attribute docstrings are ok and references PEP258 for the details: https://www.python.org/dev/peps/pep-0258/#id40

**Comment by djhoese** [comment]
Ah I missed the `in` part of that operation. Looks good (including the docstring).

**Comment by djhoese** [comment]
Is this ready for review (still a draft)?

**Comment by sfinkens** [comment]
Added a test, now ready for review!

File: satpy/composites/__init__.py [hunk 1]
```diff
@@ -334,7 +337,17 @@ def check_areas(self, data_arrays):
                       "'{}'".format(self.attrs['name']))
             raise IncompatibleAreas("Areas are different")
 
-        return data_arrays
+        # Drop neglible non-dimensional coordinates (understood to be "close-enough" for most operations)
```
Comment (mraspaud) [inline satpy/composites/__init__.py]: Could this be put in a separate function ? This looks like it adds a new functionality to the `check_areas` function, making it accomplish 2 tasks (as hinted by the docstring).

Comment (djhoese) [reply satpy/composites/__init__.py]: The problem with removing this from `check_areas` is that all composites would then have to call `check_areas` *and* this new method. I'm ok if `check_areas` also does this operation for now although moving it to a separate function wouldn't be a bad idea as long as `check_areas` calls it.

Comment (mraspaud) [reply satpy/composites/__init__.py]: Ok, I looked more at this and I believe we need a little refactoring. What we really to in `check_areas` is match the different projectables together. For that we are now checking that the areas match, and we want to add dropping non-dimensional coordinates. Hence I would suggest having as main function something called eg `match_dataarrays`, that in turn calls `check_areas`, and calls a new function called eg `drop_non_dimensional_coordinates`.

File: satpy/composites/__init__.py [hunk 2]
```diff
@@ -309,10 +312,28 @@ def apply_modifier_info(self, origin, destination):
...
+            else:
+                new_arrays.append(ds)
+
+        return new_arrays
+
     def check_areas(self, data_arrays):
         """Check that the areas of the *data_arrays* are compatible."""
         if len(data_arrays) == 1:
-            return data_arrays
+            return
```
Comment (djhoese) [inline satpy/composites/__init__.py]: These return values in `check_areas` need to be put back unless we are breaking backwards compatibility for everyone who has a custom composite outside of Satpy.

Comment (mraspaud) [reply satpy/composites/__init__.py]: how about we rename `check_areas` to `_check_areas`, and make a new `check_areas` that calls `match_data_arrays` and issues a deprecation warning ?

Comment (djhoese) [reply satpy/composites/__init__.py]: I'd be ok with that. Or, given how we may move away from area definition in the long term, call it something like `check_geolocation` or `check_coordinates`?

Comment (mraspaud) [reply satpy/composites/__init__.py]: Fine by me (you mean instead of `_check_areas` right ?)

Comment (djhoese) [reply satpy/composites/__init__.py]: Yes. So `check_areas` calls `check_geolocation` and `match_data_arrays` calls `check_geolocation`.

Comment (mraspaud) [reply satpy/composites/__init__.py]: hmm, then we need to reintroduce the return values. I was suggesting the new `check_areas` would call `match_data_arrays`...

Comment (djhoese) [reply satpy/composites/__init__.py]: 🤦‍♂ Oh yeah, duh. That makes sense. So `match_data_arrays` calls `check_geolocation` and `drop_coordinates`. `check_areas` calls `match_data_arrays` with a deprecation warning.

**Comment by gerritholl** [comment]
Unless I'm missing something, this would seem to render `check_times`, introduced by @djhoese in https://github.com/sfinkens/satpy/commit/c1b65cf18b072915f53ba58cd16bcfd613072f68, to be essentially dead code.  For https://github.com/pytroll/satpy/pull/3168 I'm trying to find a way to keep the time coordinate for composites and found that @djhoese already solved this problem in 2018, but time coordinates are apparently dropped before Daves code is reached in the first place.

**Comment by gerritholl** [comment]
I am/was indeed missing something, as in some cases, drop coordinates does not drop the coordinates, see #3353 for an example.

**Pull request merged**